contracts/airdropsSMCampaign/ExistingTokenDropSMCampaign.sol

```diff
@@ -0,0 +1,145 @@
+import "../interfaces/AirdropMerkleProof.sol";
+import "../interfaces/IAirBroFactory.sol";
+
+/// @title Airdrops existing ERC20 tokens for airdrop recipients
+contract ExistingTokenDropSMCampaign is AirdropInfoSMCampaign, AirdropMerkleProof, Ownable {
+    ERC1155 public immutable rewardedNft;
+    IERC20 public immutable rewardToken;
+    uint256 public immutable tokensPerClaim;
+    uint256 public immutable totalAirdropAmount;
+
+    bool public airdropFunded = false;
+    uint256 public airdropFundBlockTimestamp;
+    address internal airdropFundingHolder;
+    address public immutable airBroFactorySMCampaignAddress;
+
+    event Claimed(address indexed claimer);
+    event AirdropFunded(address contractAddress);
+    event MerkleRootChanged(bytes32 merkleRoot);
+
+
+    error NotOwner();
+    error AirdropStillInProgress();
+    error AlreadyRedeemed();
+    error AlreadyFunded();
+    error InsufficientAmount();
+    error InsufficientLiquidity();
+    error AirdropExpired();
+    error NotAdmin();
+    error NotEligible();
+
+    mapping(address => bool) public hasClaimed;
+
+    // The root hash of the Merle Tree we previously generated in our JavaScript code. Remember
+    // to provide this as a bytes32 type and not string. Ox should be prepended.
+    bytes32 public merkleRoot;
+
+    uint256 public immutable airdropDuration;
+    uint256 public immutable airdropStartTime;
+    uint256 public immutable airdropFinishTime;
+
+    modifier onlyAdmin(){
+        if(msg.sender != IAirBroFactory(airBroFactorySMCampaignAddress).admin()) revert NotAdmin();
+        _;
+    }
+
+    constructor(
+        address _rewardedNft,
+        uint256 _tokensPerClaim,
+        address _rewardToken,
+        uint256 _totalAirdropAmount,
+        uint256 _airdropDuration,
+        address _airBroFactorySMCampaignAddress
+    ) {
+        rewardedNft = ERC1155(_rewardedNft);
+        tokensPerClaim = _tokensPerClaim;
+        totalAirdropAmount = _totalAirdropAmount;
+        rewardToken = IERC20(_rewardToken);
+        airdropDuration = _airdropDuration * 1 days;
+        airdropStartTime = block.timestamp;
+        airdropFinishTime = block.timestamp + airdropDuration;
+        airBroFactorySMCampaignAddress = _airBroFactorySMCampaignAddress;
+    }
+
+    /// @notice Sets the merkleRoot - can only be done if admin (different from the contract owner)
+    /// @param _merkleRoot - The root hash of the Merle Tree
+    function setMerkleRoot(bytes32 _merkleRoot) external onlyAdmin {
+        merkleRoot = _merkleRoot;
+        emit MerkleRootChanged(_merkleRoot);
+    }
+
+    /// @notice Allows the airdrop creator to provide funds for airdrop reward
+    function fundAirdrop() external {
+        if (rewardToken.balanceOf(msg.sender) < totalAirdropAmount) revert InsufficientAmount();
+        if (airdropFunded) revert AlreadyFunded();
+        rewardToken.transferFrom(msg.sender, address(this), totalAirdropAmount);
```

Comment:
Also comments from the 1155NFTDropCampaign apply here
Use https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#SafeERC20